app/models/link_graph.rb

```diff
@@ -1,25 +1,41 @@
+# This class is used to represent the graph of links for an edition.
+# It is used in [link expansion](../doc/link-expansion.md) and
+# [dependency resolution](../doc/depedency-resolution.md)
 class LinkGraph
-  attr_reader :root_content_id, :link_reference
+  attr_reader :root_content_id, :root_locale, :with_drafts, :link_reference
 
-  def initialize(root_content_id, link_reference)
+  # link_reference is an object that can be queried to determine the links
+  # for given inputs. It is the source of information for representing this
+  # graph.
+  def initialize(
+    root_content_id:,
+    root_locale:,
+    with_drafts: false,
+    link_reference:
+  )
     @root_content_id = root_content_id
+    @root_locale = root_locale
+    @with_drafts = with_drafts
     @link_reference = link_reference
   end
 
+  # An array of LinkGraph::Node objects
```

Comment:
@carlosmartinez I added the comments in those classes to make it so people who weren't that familiar with them could use them to work out what's going on with them to try debug link expansion / dependency resolution. They are inconsistent so I agree with your comment but also wondered, as someone who's not as familiar with those classes, are they usable to an outsider - do they make any sense etc?